### PR TITLE
Fixes 2311, validates 20 bytes Yubikey secret

### LINKED ISSUE #2311: Poor input validation on Yubikey HOTP setup
# Bug

## Description

When configuring a Yubikey for HOTP 2FA, the `./manage.py` script does not valid input properly. 
## Steps to Reproduce

For instance, a hardware Yubikey with secret key:

> cb a0 5f ad 41 a2 ff 4e eb 53 56 3a 1b f7 23 2e ce fc dc d7

that entire secret should be pasted as-is into the prompt **Please configure your YubiKey and enter the secret:**. However, if you make a mistake copy/pasting, and truncate the string, e.g.:

> cb a0 5f ad 41 a2 ff 4e eb 53 56 3a 1b f7

Then the script will still accept the bad value, and logins will be impossible with the newly created account. 
## Expected Behavior

`./manage.py` script fails if HOTP secret input is bad, forcing re-entry.

## Actual Behavior

`./manage.py` shows a lackadaisical attitude toward HOTP secret structure, then staunchly refuses logins using the Yubikey.

## Comments

Logs show, predictably, a bad token error:

>  ERROR in journalist: Login for 'conor4' failed: invalid token

### PR DESCRIPTION
**Pull request by kushaldas**
## Status

Ready for review

## Description of Changes

Fixes #2311 

Checking the length of the secret (20 bytes) before accepting a secret key in the manage.py

## Testing

How should the reviewer test this PR?

Try to add a new user with an invalid key (length wise).

## Deployment

Any special considerations for deployment? Consider both:

1. Upgrading existing production instances.
2. New installs.

### If you made changes to the app code:

- [x] Unit and functional tests pass on the development VM

### If you made changes to the system configuration:

- [ ] [Configuration tests](https://docs.securedrop.org/en/latest/development/testing_configuration_tests.html) pass

### If you made changes to documentation:

- [ ] Doc linting passed locally

### CODE REVIEW

**Comment by kushaldas** [comment]
Should not push code without enough coffee.

File: securedrop/manage.py [hunk 1]
```diff
@@ -159,6 +159,12 @@ def _add_user(is_admin=False):
             otp_secret = raw_input(
                 "Please configure this user's YubiKey and enter the secret: ")
             if otp_secret:
+                tmp_str = otp_secret.replace(" ", "")
+                if len(tmp_str) != 40:
+                    print("The length of the secret is not correct. "
+                          "Try again.")
```
Comment (dachary) [inline securedrop/manage.py]: It would be very useful to mention the length of the secret provided and the expected length.

Comment (kushaldas) [reply securedrop/manage.py]: Will add that.

**Comment by dachary** [comment]
would you mind squashing the commits ? it would make the git history easier to read.

File: securedrop/manage.py [hunk 2]
```diff
@@ -159,6 +159,13 @@ def _add_user(is_admin=False):
             otp_secret = raw_input(
                 "Please configure this user's YubiKey and enter the secret: ")
             if otp_secret:
+                tmp_str = otp_secret.replace(" ", "")
+                if len(tmp_str) != 40:
+                    print("The length of the secret is not correct. "
+                          "Expected 40 characters, but, received {0}. "
```
Comment (redshiftzero) [inline securedrop/manage.py]: Nit: `, but,` -> `but`

**Pull request merged**